### PR TITLE
The initial Dimensions Adapter for Credbull.

### PR DESCRIPTION
**Pull request by jplodge-pro**
### To Do
- [x] Capture CBL Daily Volumes on the Aggregator Dashboard.

### Note
Hi reviewer, 
If you think **anything** is wrong/questionable with this, please let me know. Like, for example, maybe it should be in `dexs`? Or using `getLogs` is inefficient? Anything at all! Thanks! :+1:

### CODE REVIEW

**Comment by llamatester** [comment]
The credbull adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts aggregators credbull
    
    🦙 Running CREDBULL adapter 🦙
    ---------------------------------------------------
    Start Date:	Thu, 07 Nov 2024 07:10:41 GMT
    End Date:	Fri, 08 Nov 2024 07:10:41 GMT
    ---------------------------------------------------

File: aggregators/credbull/index.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const CBL_TRANSFER_EVENT_ABI = "event Transfer(address indexed from, address indexed to, uint256 value)";
+
+async function fetch(options: FetchOptions): Promise<FetchResultVolume> {
+  const dailyVolume = options.createBalances();
+  const logs = await options.getLogs({
+    target: CBL_TOKEN,
+    eventAbi: CBL_TRANSFER_EVENT_ABI,
+  });
+  logs.map((e: any) => dailyVolume.addToken(CBL_TOKEN, e.value));
+  return { dailyVolume, timestamp: Date.now() / 1000 };
```
Comment (g1nt0ki) [inline aggregators/credbull/index.ts:15]: can skip the timestamp field here

File: aggregators/credbull/index.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+const CBL_TGE = 1730973600; // 2024-11-07 10:00 UTC
+const CBL_TOKEN = "0xD6b3d81868770083307840F513A3491960b95cb6";
+const CBL_TRANSFER_EVENT_ABI = "event Transfer(address indexed from, address indexed to, uint256 value)";
+
+async function fetch(options: FetchOptions): Promise<FetchResultVolume> {
+  const dailyVolume = options.createBalances();
+  const logs = await options.getLogs({
+    target: CBL_TOKEN,
+    eventAbi: CBL_TRANSFER_EVENT_ABI,
```
Comment (g1nt0ki) [inline aggregators/credbull/index.ts:12]: why are you treating  all CBL token transfers as your app volume?

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
.

File: aggregators/credbull/index.ts (hunk 1 above)
Comment (jplodge-pro) [reply aggregators/credbull/index.ts:15]: I am explicitly returning a `Promise<FetchResultVolume>` and that requires the `timestamp`. 
It's true if I return a `Promise<any>`, or a more general return type, then the timestamp is not required. I'll change to that.

File: aggregators/credbull/index.ts (hunk 2 above)
Comment (jplodge-pro) [reply aggregators/credbull/index.ts:12]: * I'm pretty n00b, so 'all transfers' seems like the volume of the token to me. What is a better definition? Include mint/burn? I'll look further.
* I wanted to get some expert feedback on review, to make sure I was on the right track. 
* The docs are sparse, so I was trying to figure out the means of interrogating the chain, via `getLogs`.
* I haven't checked the other methods (`sdk.graph` etc.)

That said, after team discussions, I will change this to focus on our staking vault, not token. Will close PR, as not correct at all.

**Comment by jplodge-pro** [comment]
Closing, as change will be entirely different,

**Pull request closed without merging**